### PR TITLE
Add devbox

### PR DESCRIPTION
**Pull request by Diogenesoftoronto**
This is a minimal devbox setup for our project. @latest for the versions
probably does not make sense past this initial concept as that will mean
different versions of the software when people start the project.

Adding installations instruction so that people can get started with devbox is another thing that can be added.

ref #823

### CODE REVIEW

File: devbox.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "packages": [
+    "go_1_20@latest",
```
Comment (sevein) [inline devbox.json]: Shouldn't we be using "go_1_21@latest"?

File: devbox.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    "kubectl@latest",
+    "tilt@latest",
+    "k3d@latest",
```
Comment (sevein) [inline devbox.json:5]: Curious to know, what's the limit here? You list k3d, but why not docker or vscode? Genuinely asking!

File: docs/src/dev-manual/devel.md [hunk 3]
```diff
@@ -1,10 +1,15 @@
 # Local/Development environment
 
-Enduro uses Tilt to set up a local environment building the Docker images in a
-Kubernetes cluster. It has been tested with k3d, Minikube and Kind.
+## Quickstart with Devbox
+
+First install [devbox]. After installing devbox, run `devbox shell`. Continue
+the instructions [here](#dex-host).
```
Comment (sevein) [inline docs/src/dev-manual/devel.md]: Can you make this a subsection of Requirements and briefly describe what devbox is and that it's experimental and not part of the standard dev workflow yet?

**Review by sevein — CHANGES_REQUESTED** [review]
Thanks for doing this! I've always wanted to try devbox. It has potential to replace the hack/make stuff we use to build our tools.

I've noticed that you haven't included the lock file, is there any reason for not doing that? It feels like it's going to be important to have a shared lock file for reproducibility, just like our go.sum, etc.

File: devbox.json (hunk 1 above)
Comment (Diogenesoftoronto) [reply devbox.json]: Yeah honestly I think I just forgot go1.21 released

File: devbox.json (hunk 2 above)
Comment (Diogenesoftoronto) [reply devbox.json:5]: docker is a good addition, however not sure if we should package vscode, as not everyone uses the editor. I was trying get a minimal set of tools need for the project to work. Other developer tools like k9s, github-cli, glow, and fish or useful scripts that we like beyond make are also options.

**Comment by Diogenesoftoronto** [comment]
> Thanks for doing this! I've always wanted to try devbox. It has potential to replace the hack/make stuff we use to build our tools.
> Yes it is very useful for that. There is a lot to explore.
> I've noticed that you haven't included the lock file, is there any reason for not doing that? It feels like it's going to be important to have a shared lock file for reproducibility, just like our go.sum, etc.

I considered adding the lockfile, and my initial belief about the lockfile was wrong. I am going to add it.

File: docs/src/dev-manual/devel.md [hunk 4]
```diff
@@ -1,10 +1,21 @@
...
 
+## Requirements
+
+### Quickstart with Devbox
+
+Devbox is currently an experimental approach to managing and sharing our
+developer environments quickly and efficiently, it is not however _well
+tested_.
+
+First install [devbox]. After installing devbox, run `devbox shell`. Continue
```
Comment (sevein) [inline docs/src/dev-manual/devel.md]:
```suggestion
First install [Devbox]. After installing Devbox, run `devbox shell`. Continue
```

File: docs/src/dev-manual/devel.md [hunk 5]
```diff
@@ -1,10 +1,21 @@
...
+the instructions [here](#dex-host).
+
+### Standard Installation
```
Comment (sevein) [inline docs/src/dev-manual/devel.md]: We're using sentence case capitalization:

```suggestion
### Standard installation
```

File: docs/src/dev-manual/devel.md [hunk 6]
```diff
@@ -247,3 +258,4 @@ is sometimes not setup properly. To solve it, from the Tilt UI, restart the
 [gcc]: https://gcc.gnu.org/
 [visual studio code]: https://code.visualstudio.com/
 [working with archivematica]: archivematica.md
+[devbox]: https://www.jetpack.io/devbox/docs/quickstart/#install-devbox
```
Comment (sevein) [inline docs/src/dev-manual/devel.md:261]:
```suggestion
[Devbox]: https://www.jetpack.io/devbox/docs/quickstart/#install-devbox
```

**Review by sevein — APPROVED** [review]
LGTM, only a few minor changes requested.

File: docs/src/dev-manual/devel.md (hunk 6 above)
Comment (Diogenesoftoronto) [reply docs/src/dev-manual/devel.md:261]: hmm this seems out of alighnment with how we have capitalised every other variable here.

Comment (sevein) [reply docs/src/dev-manual/devel.md:261]: Oh maybe it's case insensitive? That makes sense. I only meant that the proper name is Devbox as opposed to devbox.

**Pull request merged**